### PR TITLE
Adds Redactive python SDK

### PR DESCRIPTION
**Pull request by russellwmy**
Changes
- Added Python SDK
- Auto generate python sdk gRpc
- Configured pre-commit hooks for unit testing, formatting and commit message format check

Tests
- Run `pre-commit run --all` with devcontainer

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -13,14 +14,62 @@ repos:
...
+  # Python SDK hooks
+  - repo: local
+    hooks:
+      # Install Python SDK
+      - id: python-sdk-install
+        name: Install python SDK
+        language: python
+        types: [python]
+        # Custom command for monorepo
+        entry: bash -c 'cd sdks/python && pip install .[dev,test]'
```
Comment (MattTimms) [inline .pre-commit-config.yaml]: I have not been able to get this step running successfully.

```
      ...
      error: [Errno 17] File exists: 'build/bdist.macosx-14.4-arm64/wheel/redactive-0.0.5.dist-info'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for redactive
Failed to build redactive
ERROR: Could not build wheels for redactive, which is required to install pyproject.toml-based projects
```

Comment (russellwmy) [reply .pre-commit-config.yaml]: hm.. The package install in pre-commit env executes differently. Let me check first.

**Review by MattTimms — APPROVED** [review]
LGTM! Could we, in this or the next PR, uncomment `.idea` & `.python-version` from `.gitignore`?

**Comment by russellwmy** [comment]
> LGTM! Could we, in this or the next PR, uncomment `.idea` & `.python-version` from `.gitignore`?

I am create new PR just for that.

**Pull request merged**